src/Quizzer.API.DAL/Repository/Repository.cs

```diff
@@ -55,10 +59,18 @@ public class RepositoryApiBase<TEntity> : IRepository<TEntity>, IDisposable
 
         public Guid Insert(TEntity entity) => DbSet().Add(entity).Entity.Id;
 
-        public Guid? Update(TEntity entity) {
+        public Guid? Update(TEntity entity, string navigationPropertyPath = "") {
+            return Update(entity, new string[] { navigationPropertyPath });
+        }
+
+        public Guid? Update(TEntity entity, IEnumerable<string> navigationPropertyPaths) {
             if (!Exists(entity.Id)) return null;
 
-            DbSet().Attach(entity);
```

Comment:
Why the attach was removed?

Reply:
Ok, what function did attach do?

Reply:
Attach made sure it is tracked by the EF, but if you call GetById right before it should also be tracked.

Reply:
I thought because of AutoMapper null properties, but it should not mapper since it's destination anyway.
TBH I don't understand why they do it like this in CB for recipies. I think Repository should work the way it is, How about we leave this PR right now and come back to it if Repository test's will fail?